### PR TITLE
[usage] Ignore running instances before August 2022

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

We've got bad data quality on workspace_instances from the past. This excludes these from our queries for running workspaces until we can fix the source of truth.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Review by svenefftinge — APPROVED** [review]

File: components/usage/pkg/db/workspace_instance.go [hunk 1]
```diff
@@ -80,6 +80,8 @@ func FindRunningWorkspaceInstances(ctx context.Context, conn *gorm.DB) ([]Worksp
 	tx := queryWorkspaceInstanceForUsage(ctx, conn).
 		Where("wsi.stoppingTime = ?", "").
 		Where("wsi.usageAttributionId != ?", "").
+		// Exclude anything before August 2022, we cannot guarantee data quality before then
+		Where("wsi.startedTime > ?", TimeToISO8601(time.Date(2022, 8, 1, 0, 0, 0, 0, time.UTC))).
```
Comment (jankeromnes) [inline components/usage/pkg/db/workspace_instance.go:84]: Question: Wasn't this part of why we have a `billInstancesAfter`? 💭

Comment (svenefftinge) [reply components/usage/pkg/db/workspace_instance.go:84]: `billInstancesAfter` would be a check against the `effectiveTime`. Running Instances have their `effectiveTime` now or in the future.

Comment (easyCZ) [reply components/usage/pkg/db/workspace_instance.go:84]: The `billInstancesAfter` is/was used in the BillingService. We could re-use it here as well

Comment (svenefftinge) [reply components/usage/pkg/db/workspace_instance.go:88]: I don't think so. It should actually be named `billInstancesStoppedAfter` to make clear it has nothing to do with when a workspace was started.

Comment (svenefftinge) [reply components/usage/pkg/db/workspace_instance.go:84]: I think we should remove `billInstancesStoppedAfter` entirely. Doesn't make sense to have a global starting point. Instead we should have the information how usage gets billed associated with each cost center. I'm going to work on this in the context of https://github.com/gitpod-io/gitpod/issues/12741

**Pull request merged**